File: aux_music/test_inv.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final instances = [
    'invidious.jing.rocks',
    'inv.tux.pizza',
    'invidious.nerdvpn.de',
  ];
  
  final videoId = 'kPa7bsKwL-c';
  
  for (final instance in instances) {
    try {
      print('Testing $instance...');
      final client = HttpClient();
      final request = await client.getUrl(Uri.parse('https://$instance/api/v1/videos/$videoId'));
      final response = await request.close().timeout(Duration(seconds: 5));
      if (response.statusCode == 200) {
        final body = await response.transform(utf8.decoder).join();
        final json = jsonDecode(body);
        print('SUCCESS: $instance works! length: ');
      } else {
        print('FAILED: $instance (Status )');
      }
    } catch (e) {
      print('FAILED: $instance (Error: $e)');
    }
  }
}

File: bff/test_inv.js
const fetch = require('axios');
async function test() {
  try {
    const res = await fetch('https://api.invidious.io/instances.json');
    const instances = res.data;
    const working = instances.filter(i => i[1].type === 'https' && i[1].api && i[1].cors);
    working.sort((a,b) => a[1].health - b[1].health);
    for (let i=0; i<3; i++) {
      const uri = working[i][1].uri;
      try {
        const testRes = await fetch(uri + '/api/v1/videos/dQw4w9WgXcQ', { timeout: 3000 });
        if (testRes.status === 200) console.log(uri + ' works!');
      } catch(e) {}
    }
  } catch(e) { console.error(e.message); }
}
test();
